examples/prom/maxwell_global_rom.cpp

```diff
@@ -0,0 +1,548 @@
+
+        // 21. form inverse ROM operator
+        CAROM::Matrix invReducedA(numColumnRB, numColumnRB, false);
+        ComputeCtAB( *A.As<HypreParMatrix>(), *spatialbasis, *spatialbasis,
+                     invReducedA);
+        invReducedA.inverse();
+
+        CAROM::Vector B_carom(B.GetData(), B.Size(), true, false);
+        CAROM::Vector X_carom(X.GetData(), X.Size(), true, false);
+        CAROM::Vector *reducedRHS = spatialbasis->transposeMult(&B_carom);
+        CAROM::Vector reducedSol(numColumnRB, false);
+        assembleTimer.Stop();
+
+        // 22. solve ROM
+        solveTimer.Start();
+        invReducedA.mult(*reducedRHS, reducedSol);
+        solveTimer.Stop();
+
+        // 23. reconstruct FOM state
+        spatialbasis->mult(reducedSol, X_carom);
+        delete spatialbasis;
+        delete reducedRHS;
+    }
+
+    // 24. Recover the parallel grid function corresponding to X. This is the
+    //     local finite element solution on each processor.
+    a->RecoverFEMSolution(X, *b, x);
+
+    // 26. Calculate the relative error of the ROM prediction compared to FOM
+    ostringstream sol_dofs_name, sol_dofs_name_fom;
+    if (fom || offline)
+    {
+        sol_dofs_name << "sol_dofs_fom." << setfill('0') << setw(6) << myid;
+    }
+    if (online)
+    {
+        sol_dofs_name << "sol_dofs." << setfill('0') << setw(6) << myid;
+        sol_dofs_name_fom << "sol_dofs_fom." << setfill('0') << setw(6) << myid;
+    }
+
+    if (online)
+    {
+        // Initialize FOM solution
+        Vector x_fom(x.Size());
+
+        ifstream fom_file;
+
+        // Open and load file
+        fom_file.open(sol_dofs_name_fom.str().c_str());
+
+        x_fom.Load(fom_file, x_fom.Size());
+
+        fom_file.close();
+
+        Vector diff_x(x.Size());
+
+        subtract(x, x_fom, diff_x);
+
+        // Get norms
+        double tot_diff_norm = sqrt(InnerProduct(MPI_COMM_WORLD, diff_x, diff_x));
+
+        double tot_fom_norm = sqrt(InnerProduct(MPI_COMM_WORLD,
+                                                x_fom, x_fom));
+
+        if (myid == 0)
+        {
+            cout << "Relative error of ROM solution = "
+                 << tot_diff_norm / tot_fom_norm << endl;
+        }
+    }
+
+    // 27. Save the refined mesh and the solution in parallel. This output can
+    //     be viewed later using GLVis: "glvis -np <np> -m mesh -g sol".
+    {
+        ostringstream mesh_name, sol_name;
+        mesh_name << "mesh." << setfill('0') << setw(6) << myid;
+        sol_name << "sol." << setfill('0') << setw(6) << myid;
+
+        ofstream mesh_ofs(mesh_name.str().c_str());
+        mesh_ofs.precision(8);
+        pmesh->Print(mesh_ofs);
+
+        ofstream sol_ofs(sol_name.str().c_str());
+        sol_ofs.precision(8);
+        x.Save(sol_ofs);
+    }
+
+    // 28. Save data in the VisIt format.
+    DataCollection *dc = NULL;
+    if (visit)
+    {
+        if (offline) dc = new VisItDataCollection("Example3", pmesh);
```

Comment:
Since the name of the example has been changed from `ex3p` to `maxwell...`, this name `Example3` could be changed to `maxwell`.

Reply:
Thanks Dylan, changed the names of these folders to ```maxwell```.